### PR TITLE
Setup BDJuno

### PR DESCRIPTION
**Pull request by hochiw**
## Features 

- Setup BDJuno as a submodule for likedao
- Separated BDJuno database and graphql server database for local development
- Added bdjuno database to graphql server context
- Setup makefiles and docker-compose service for bdjuno
- Updated README.md
- Added a query for fetching latest block information from the bdjuno db
```
query {
  getLatestBlock {
    height
    hash
    numOfTxs
    totalGas
    proposerAddress
    timestamp
  }
}
```

### Note: The iscn module for bdjuno is temporarily disabled and master branch is checked out, this is due to the likecoin branch using v0.42 cosmos sdk which is incapable of parsing chain data that are generated via v0.44 sdk 

based on #15 
refs #3

### CODE REVIEW

File: graphql-server/pkg/mutators/test.go [hunk 1]
```diff
@@ -12,20 +12,20 @@ type ITestMutator interface {
 }
 
 type TestMutator struct {
-	ctx     context.Context
-	session *bun.DB
+	ctx           context.Context
+	serverSession *bun.DB
 }
```
Comment (chihimng) [inline graphql-server/pkg/mutators/test.go:17]: no need to rename if this only concerns one db?

File: graphql-server/pkg/queries/test.go [hunk 2]
```diff
@@ -12,17 +12,17 @@ type ITestQuery interface {
 }
 
 type TestQuery struct {
-	ctx     context.Context
-	session *bun.DB
+	ctx           context.Context
+	serverSession *bun.DB
```
Comment (chihimng) [inline graphql-server/pkg/queries/test.go:16]: no need to rename if this only concerns one db?

File: bdjuno/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# BDJuno
+
+BDJuno is used as a crawler to synchonize in-chain data to a database
```
Comment (chihimng) [inline bdjuno/README.md:3]: on-chain

File: bdjuno/Makefile [hunk 4]
```diff
@@ -4,8 +4,12 @@ GENESIS_URL?=
 .checkenv:
 	@test $${GENESIS_URL?Please set argument GENESIS_URL}
 
+.PHONY: init
+init: 
+	git submodule update --init --recursive
```
Comment (chihimng) [inline bdjuno/Makefile:9]: 👍

**Review by chihimng — CHANGES_REQUESTED** [review]
can we have a quick demo on querying data from chain data db? say get the current sync progress?

**Comment by hochiw** [comment]
@chihimng Updated :adore2x:

**Comment by chihimng** [comment]
@hochiw how about:

> can we have a quick demo on querying data from chain data db? say get the current sync progress?

**Comment by hochiw** [comment]
@chihimng added a demo query for this

I have also temporarily locked the submodule to the latest commit in [aaron's dual prefix branch](https://github.com/forbole/bdjuno/commits/aaron/likecoin_dual_prefix) just so it doesn't crash when processing a block involving validator operations, or else it would never be able to catch up to latest blocks. Will open an issue to handle pointing the submodule to a stable build when it's available.

**Review by chihimng — APPROVED** [review]
👍

**Pull request merged**